cmd/encryption-v1.go

```diff
@@ -238,7 +245,7 @@ func rotateKey(oldKey []byte, newKey []byte, metadata map[string]string) error {
 		return errObjectTampered
 	}
 	iv, err := base64.StdEncoding.DecodeString(metadata[ServerSideEncryptionIV])
-	if err != nil || len(iv) != 32 {
+	if err != nil || len(iv) != SSECustomerKeySize {
```

Comment:
Maybe add a comment that the `iv` must not be of the same size as the key. It's just chosen to be 256 bits so that selecting an `iv` twice is an **very** unlikely event. Currently it looks like the `iv` must be of the same size as the key which is not the case...

Reply:
> Maybe add a comment that the iv must not be of the same size as the key. It's just chosen to be 256 bits so that selecting an iv twice is an very unlikely event. Currently it looks like the iv must be of the same size as the key which is not the case...

I added a check for IV size to check if it is tampered.. but it is only backward compatible. (Minio doesn't have a way to check if Minio server version is less than data version)